src/simd.jl

```diff
@@ -0,0 +1,576 @@
+# this files contains some performance related stuff
+module ST # SIMDTensors
+
+using Tensors
+using Tensors: get_data
+using Compat
+
+import SIMD
+@compat const SVec{N, T} = SIMD.Vec{N, T}
+
+const SIMDTypes = Union{Bool,
+                        Int8, Int16, Int32, Int32, Int128,
+                        UInt8, UInt16, UInt32, UInt32, UInt128,
+                        Float16, Float32, Float64}
+
+# SIMD sizes accepted by LLVM between 1 and 100
+const SIMD_CHUNKS = (1, 2, 3, 4, 5, 6, 8, 9, 10, 12, 16, 17, 18, 20, 24, 32, 33, 34, 36, 40, 48, 64, 65, 66, 68, 72, 80, 96)
+
+# factors for the symmetric tensors
+function symmetric_factors(order, dim, T)
+    if order == 2
+        dim == 1 && return SVec{1, T}((T(1),))
+        dim == 2 && return SVec{3, T}((T(1),T(2),T(1)))
+        dim == 3 && return SVec{6, T}((T(1),T(2),T(2),T(1),T(2),T(1)))
+    elseif order == 4
+        dim == 1 && return SVec{1, T}((T(1),))
+        dim == 2 && return SVec{9, T}((T(1),T(2),T(1),T(2),T(4),T(2),T(1),T(2),T(1)))
+        dim == 3 && return SVec{36,T}((T(1),T(2),T(2),T(1),T(2),T(1),T(2),T(4),T(4),T(2),T(4),T(2),
+                                       T(2),T(4),T(4),T(2),T(4),T(2),T(1),T(2),T(2),T(1),T(2),T(1),
+                                       T(2),T(4),T(4),T(2),T(4),T(2),T(1),T(2),T(2),T(1),T(2),T(1)))
+    end
+end
+
+# norm
+@inline function norm{dim, T <: SIMDTypes}(S::Union{Vec{dim, T}, Tensor{2, dim, T}, Tensor{4, dim, T}})
+    v = zero(T)
+    @inbounds @simd for i in 1:length(S)
+        v += S[i] * S[i]
+    end
+    return sqrt(v)
+end
+
+@inline function norm{T <: SIMDTypes}(S::Tensor{1, 1, T})
```

Comment:
No. Just that some of these (in particular all methods for dim = 1) probably should be removed. It results in the same code anyway.